client/admin/sidebar/useSettingsGroupsFiltered.js

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useDebouncedValue } from '@rocket.chat/fuselage-hooks';
+
+import { PrivateSettingsCachedCollection } from '../PrivateSettingsCachedCollection';
+import { useTranslation } from '../../contexts/TranslationContext';
+import { settings } from '../../../app/settings';
+
+
+export const useSettingsGroupsFiltered = (textFilter) => {
```

Comment:
Most of the contents of this hook is similar to code used in `SettingsState`, right? Is there a way to share some code between then? Especially the `PrivateSettingsCachedCollection` initialization, as I think it must be a singleton.